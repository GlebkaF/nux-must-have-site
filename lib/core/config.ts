import { noisegate } from "lib/core/blocks/noisegate";
import { compressor } from "./blocks/compressor";
import { BlockConfig, Blocks } from "./interface";
import { modulation } from "lib/core/blocks/modulation";
import { effect } from "lib/core/blocks/effect";
import { amplifier } from "lib/core/blocks/amplifier";
import { eq } from "lib/core/blocks/eq";
import { cabinet } from "lib/core/blocks/cabinet";
import { reverb } from "lib/core/blocks/reverb";
import { delay } from "lib/core/blocks/delay";

interface ChainConfig {
  [key: string]: BlockConfig;
}

export const config: ChainConfig = {
  [Blocks.Noisegate]: noisegate,
  [Blocks.Compressor]: compressor,
  [Blocks.Modulation]: modulation,
  [Blocks.Effect]: effect,
  [Blocks.Amplifier]: amplifier,
  [Blocks.Cabinet]: cabinet,
  [Blocks.Eq]: eq,
  [Blocks.Reverb]: reverb,
  [Blocks.Delay]: delay,
};
