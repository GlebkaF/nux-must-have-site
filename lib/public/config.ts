export const publicConfig = {
  contacts: {
    email: "request@nxrig.com",
    telegram: "https://t.me/+JUGLVuLHoKgzZWZi",
  },
  email: {
    subjects: {
      presetRequest: "nxrig.com - Preset Request",
    },
    templates: {
      presetRequest: (songName: string) =>
        `I would like to request a preset for the song ${songName}`,
    },
  },
} as const;
