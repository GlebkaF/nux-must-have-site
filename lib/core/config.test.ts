import { describe, expect, test } from "vitest";
import { config } from "./config";

describe("Config", () => {
  test("compressor should to match snapshot", () => {
    expect(config).toMatchSnapshot();
  });
});
