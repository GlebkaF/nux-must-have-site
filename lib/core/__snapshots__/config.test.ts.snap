// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Config > compressor should to match snapshot 1`] = `
{
  "amplifier": {
    "types": [
      {
        "encodeType": 1,
        "label": "Jazz Clean",
        "params": [
          {
            "encodeIndex": 27,
            "label": "Gain",
          },
          {
            "encodeIndex": 28,
            "label": "Master",
          },
          {
            "encodeIndex": 29,
            "label": "Bass",
          },
          {
            "encodeIndex": 30,
            "label": "Middle",
          },
          {
            "encodeIndex": 31,
            "label": "Treble",
          },
          {
            "encodeIndex": 32,
            "label": "Bright",
          },
        ],
        "realName": "Jazz Clean",
      },
    ],
  },
  "cabinet": {
    "types": [
      {
        "encodeType": 1,
        "label": "JZ120",
        "params": [
          {
            "encodeIndex": 78,
            "label": "Level",
          },
          {
            "encodeIndex": 79,
            "label": "LowCut",
          },
          {
            "encodeIndex": 80,
            "label": "HighCut",
          },
        ],
        "realName": "Roland JC-120 2x12",
      },
    ],
  },
  "compressor": {
    "types": [
      {
        "encodeType": 2,
        "label": "K Comp",
        "params": [
          {
            "encodeIndex": 16,
            "label": "Level",
          },
          {
            "encodeIndex": 15,
            "label": "Sustain",
          },
          {
            "encodeIndex": 17,
            "label": "Clipping",
          },
        ],
        "realName": "Keeley Compressor",
      },
      {
        "encodeType": 3,
        "label": "Studio Comp",
        "params": [
          {
            "encodeIndex": 17,
            "label": "Gain",
          },
          {
            "encodeIndex": 15,
            "label": "Threshold",
          },
          {
            "encodeIndex": 16,
            "label": "Ratio",
          },
          {
            "encodeIndex": 18,
            "label": "Release",
          },
        ],
        "realName": "Studio Compressor",
      },
      {
        "encodeType": 1,
        "label": "Rose Comp",
        "params": [
          {
            "encodeIndex": 16,
            "label": "Level",
          },
          {
            "encodeIndex": 15,
            "label": "Sustain",
          },
        ],
        "realName": "Keeley Compressor",
      },
    ],
  },
  "delay": {
    "types": [
      {
        "encodeType": 1,
        "label": "Analog Delay",
        "params": [
          {
            "encodeIndex": 63,
            "label": "Intensity",
          },
          {
            "encodeIndex": 61,
            "label": "Rate",
          },
          {
            "encodeIndex": 62,
            "label": "Echo",
          },
        ],
        "realName": "Analog Delay",
      },
    ],
  },
  "effect": {
    "types": [
      {
        "encodeType": 1,
        "label": "Distortion+",
        "params": [
          {
            "encodeIndex": 20,
            "label": "Output",
          },
          {
            "encodeIndex": 21,
            "label": "Sensitivity",
          },
        ],
        "realName": "Distortion+",
      },
    ],
  },
  "eq": {
    "types": [
      {
        "encodeType": 1,
        "label": "6-Band",
        "params": [
          {
            "encodeIndex": 36,
            "label": "100",
          },
          {
            "encodeIndex": 37,
            "label": "220",
          },
          {
            "encodeIndex": 38,
            "label": "500",
          },
          {
            "encodeIndex": 39,
            "label": "1200",
          },
          {
            "encodeIndex": 40,
            "label": "2600",
          },
          {
            "encodeIndex": 41,
            "label": "6400",
          },
        ],
        "realName": "6-Band",
      },
      {
        "encodeType": 3,
        "label": "10-Band",
        "params": [
          {
            "encodeIndex": 36,
            "label": "Vol",
          },
          {
            "encodeIndex": 37,
            "label": "31",
          },
          {
            "encodeIndex": 38,
            "label": "62",
          },
          {
            "encodeIndex": 39,
            "label": "125",
          },
          {
            "encodeIndex": 40,
            "label": "250",
          },
          {
            "encodeIndex": 41,
            "label": "500",
          },
          {
            "encodeIndex": 42,
            "label": "1000",
          },
          {
            "encodeIndex": 43,
            "label": "2000",
          },
          {
            "encodeIndex": 44,
            "label": "4000",
          },
          {
            "encodeIndex": 45,
            "label": "8000",
          },
          {
            "encodeIndex": 46,
            "label": "16000",
          },
        ],
        "realName": "10-Band",
      },
    ],
  },
  "modulation": {
    "types": [
      {
        "encodeType": 1,
        "label": "CE-1",
        "params": [
          {
            "encodeIndex": 56,
            "label": "Rate",
          },
          {
            "encodeIndex": 55,
            "label": "Depth",
          },
          {
            "encodeIndex": 54,
            "label": "Intensity",
          },
        ],
        "realName": "CE-1",
      },
    ],
  },
  "noisegate": {
    "types": [
      {
        "encodeType": 1,
        "label": "Noise Gate",
        "params": [
          {
            "encodeIndex": 49,
            "label": "Sensitivity",
          },
          {
            "encodeIndex": 50,
            "label": "Decay",
          },
        ],
        "realName": "Noise Gate",
      },
    ],
  },
  "reverb": {
    "types": [
      {
        "encodeType": 1,
        "label": "Room",
        "params": [
          {
            "encodeIndex": 72,
            "label": "Level",
          },
          {
            "encodeIndex": 70,
            "label": "Decay",
          },
          {
            "encodeIndex": 71,
            "label": "Tone",
          },
        ],
        "realName": "Room Reverb",
      },
      {
        "encodeType": 2,
        "label": "Hall",
        "params": [
          {
            "encodeIndex": 73,
            "label": "Level",
          },
          {
            "encodeIndex": 70,
            "label": "Decay",
          },
          {
            "encodeIndex": 71,
            "label": "PreDelay",
          },
          {
            "encodeIndex": 72,
            "label": "Liveliness",
          },
        ],
        "realName": "Hall Reverb",
      },
      {
        "encodeType": 3,
        "label": "Plate",
        "params": [
          {
            "encodeIndex": 71,
            "label": "Level",
          },
          {
            "encodeIndex": 70,
            "label": "Decay",
          },
        ],
        "realName": "Plate Reverb",
      },
      {
        "encodeType": 4,
        "label": "Spring",
        "params": [
          {
            "encodeIndex": 71,
            "label": "Level",
          },
          {
            "encodeIndex": 70,
            "label": "Decay",
          },
        ],
        "realName": "Spring Reverb",
      },
      {
        "encodeType": 5,
        "label": "Shimmer",
        "params": [
          {
            "encodeIndex": 70,
            "label": "Mix",
          },
          {
            "encodeIndex": 71,
            "label": "Decay",
          },
          {
            "encodeIndex": 72,
            "label": "Shimmer",
          },
        ],
        "realName": "Shimmer Reverb",
      },
      {
        "encodeType": 6,
        "label": "Damp",
        "params": [
          {
            "encodeIndex": 70,
            "label": "Mix",
          },
          {
            "encodeIndex": 71,
            "label": "Depth",
          },
        ],
        "realName": "Damp Reverb",
      },
    ],
  },
}
`;
